bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/AbstractContainerImplTest.java

```diff
@@ -0,0 +1,125 @@
+package com.adobe.cq.wcm.core.components.internal.models;
+
+import java.util.List;
+
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ValueMap;
+import org.apache.sling.api.scripting.SlingBindings;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.testing.mock.sling.servlet.MockSlingHttpServletRequest;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.sightly.WCMBindings;
+import com.adobe.cq.wcm.core.components.context.CoreComponentTestContext;
+import com.adobe.cq.wcm.core.components.models.Container;
+import com.adobe.cq.wcm.core.components.models.ListItem;
+import com.day.cq.wcm.api.designer.Style;
+import io.wcm.testing.mock.aem.junit.AemContext;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class AbstractContainerImplTest {
+
+    private static final String TEST_BASE = "/container";
+    private static final String CONTENT_ROOT = "/content";
+    private static final String CONTEXT_PATH = "/core";
+    private static final String TEST_ROOT_PAGE = "/content/container";
+    private static final String TEST_ROOT_PAGE_GRID = "/jcr:content/root/responsivegrid";
+    private static final String CAROUSEL_1 = TEST_ROOT_PAGE + TEST_ROOT_PAGE_GRID + "/container-1";
+    private static final String TEST_APPS_ROOT = "/apps/core/wcm/components";
+
+    @ClassRule
+    public static final AemContext AEM_CONTEXT = CoreComponentTestContext.createContext(TEST_BASE, CONTENT_ROOT);
+
+    @BeforeClass
+    public static void init() {
+        AEM_CONTEXT.load().json(TEST_BASE + CoreComponentTestContext.TEST_APPS_JSON, TEST_APPS_ROOT);
+    }
+
+    @Test
+    public void testEmptyContainer() {
+        Container container = new ContainerImpl();
+        List<ListItem> items = container.getItems();
+        assertTrue("", items == null || items.size() == 0);
+    }
+
+    @Test
+    public void testContainerWithItems() {
+        Container container = getContainerUnderTest(CAROUSEL_1);
+        Object[][] expectedItems = {
+            {"Teaser 1 description", "item_1", "/content/container/jcr:content/root/responsivegrid/container-1/item_1", "Teaser 1"},
+            {"Teaser 2 description", "item_2", "/content/container/jcr:content/root/responsivegrid/container-1/item_2", "Teaser 2"},
+        };
+        verifyContainerItems(expectedItems, container.getItems());
+    }
+
+    private Container getContainerUnderTest(String resourcePath) {
```

Comment:
The code might be simplified with the new version of wcm-io-testing. See e.g.:
https://github.com/adobe/aem-core-wcm-components/blob/issue/298/bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/models/TabsImplTest.java#L81